Add definition for rasterdata

jansule opened the pull request:
# Breaking Change

Introduced two new interfaces `VectorData` and `RasterData` that extend `BaseData`. Now, `Data` is either `RasterData` or `VectorData`. Updated the readFile output parameter of `DataParser` accordingly.

Due to the refactoring of `Data`, this is a breaking change.

KaiVolland commented:
Does RasterData has `exampleFeatures`? Is it enough to just add another `property` to our DataSchema?

jansule commented:
> Does RasterData has `exampleFeatures`? 
Not really, as raster data does not have any features. However, there are some properties that can be beneficial for us (bandnames, ranges, other stats).

> Is it enough to just add another `property` to our DataSchema?
It is enough for describing the properties but not for holding their values.

Rewrote the definition and now `Data` only contains `DataSchema`. Introduced two new interfaces `VectorData` and `RasterData` that extend `Data`. Will update PR description accordingly.
